file/src/main/java/org/apache/calcite/adapter/file/CsvEnumerator.java

```diff
@@ -130,30 +137,70 @@ public static RelDataType deduceRowType(JavaTypeFactory typeFactory,
       }
       for (String string : strings) {
         final String name;
-        final CsvFieldType fieldType;
+        final RelDataType fieldType;
         final int colon = string.indexOf(':');
         if (colon >= 0) {
           name = string.substring(0, colon);
           String typeString = string.substring(colon + 1);
-          fieldType = CsvFieldType.of(typeString);
-          if (fieldType == null) {
-            System.out.println("WARNING: Found unknown type: "
-                + typeString + " in file: " + source.path()
-                + " for column: " + name
-                + ". Will assume the type of column is string");
+          Matcher decimalMatcher = DECIMAL_TYPE_PATTERN.matcher(typeString);
+          if (decimalMatcher.matches()) {
+            int precision = Integer.parseInt(decimalMatcher.group(1));
+            int scale = Integer.parseInt(decimalMatcher.group(2));
+            fieldType = parseDecimalSqlType(typeFactory, precision, scale);
+          } else {
+            switch (typeString) {
+            case "string":
+              fieldType = toNullableRelDataType(typeFactory, SqlTypeName.VARCHAR);
+              break;
+            case "boolean":
+              fieldType = toNullableRelDataType(typeFactory, SqlTypeName.BOOLEAN);
+              break;
+            case "byte":
+              fieldType = toNullableRelDataType(typeFactory, SqlTypeName.TINYINT);
+              break;
+            case "char":
+              fieldType = toNullableRelDataType(typeFactory, SqlTypeName.CHAR);
+              break;
+            case "short":
+              fieldType = toNullableRelDataType(typeFactory, SqlTypeName.SMALLINT);
+              break;
+            case "int":
+              fieldType = toNullableRelDataType(typeFactory, SqlTypeName.INTEGER);
+              break;
+            case "long":
+              fieldType = toNullableRelDataType(typeFactory, SqlTypeName.BIGINT);
+              break;
+            case "float":
+              fieldType = toNullableRelDataType(typeFactory, SqlTypeName.REAL);
+              break;
+            case "double":
+              fieldType = toNullableRelDataType(typeFactory, SqlTypeName.DOUBLE);
+              break;
+            case "date":
+              fieldType = toNullableRelDataType(typeFactory, SqlTypeName.DATE);
+              break;
+            case "timestamp":
+              fieldType = toNullableRelDataType(typeFactory, SqlTypeName.TIMESTAMP);
+              break;
+            case "time":
+              fieldType = toNullableRelDataType(typeFactory, SqlTypeName.TIME);
+              break;
+            default:
+              LOGGER.warn(
+                  String.format(Locale.ROOT,
+                  "Found unknown type: %s in file: %s for column: %s. Will assume the type of "
```

Comment:
when using a logger, you should use the loggers capability for value replacement as opposed to string format.

Reply:
Done.

Reply:
Usually loggers use {}. Isn't that what should be used (instead of %s?)

Reply:
Done.